components/dashboard/src/settings/Preferences.tsx

```diff
@@ -153,6 +161,12 @@ export default function Preferences() {
                     </div>
                 </SelectableCard>
             </div>
+            <h3 className="mt-12">Dotfiles <PillLabel type="warn" className="font-semibold mt-2 py-0.5 px-2 self-center">Beta</PillLabel></h3>
+            <p className="text-base text-gray-500 dark:text-gray-400">Customise every workspace using dotfiles. Add a repo below which gets cloned and installed during workspace startup.</p>
+            <div className="mt-4">
+                <h4>Repo</h4>
+                <input type="text" value={dotfileRepo} onChange={(e) => actuallySetDotfileRepo(e.target.value)} className="w-full" />
```

Comment:
**issue(non-blocking)**: Auto-saving the input here happens in the background but is lacking :a: call to action and :b: some user feedback upon save which both will probably be resolved in the second iteration in https://github.com/gitpod-io/gitpod/issues/7385.

For :a:, we could add an action button below to save changes so that users won't be surprized when they accidentally change this input and dotfiles configuration fails, etc. Alternatively, we could introduce a new auto-save pattern that makes it visible to the user that we auto-save on every key press.

For :b:, ideally this manual save action would also need some feedback in the user interface so that users can safely acknowledge that the setting has been updated. This could be done with a subtle loading indicator (spinner) inside the button when saving changes that remains visible for a noticeable period of time or using a toast notification (https://github.com/gitpod-io/gitpod/issues/3530) which can be considered out of the scope of this PR.